File: app/src/main/java/com/OverSadBoy/samurairise/Item.java
package com.OverSadBoy.samurairise;

import android.os.Parcel;
import android.os.Parcelable;

public class Item implements Parcelable {

    private int id;
    private String time;
    private String repeat;
    private String status;

    public Item(int id, String time, String repeat, String status) {
        this.id = id;
        this.time = time;
        this.repeat = repeat;
        this.status = status;
    }

    protected Item(Parcel in) {
        id = in.readInt();
        time = in.readString();
        repeat = in.readString();
        status = in.readString();
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getRepeat() {
        return repeat;
    }

    public void setRepeat(String repeat) {
        this.repeat = repeat;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(id);
        parcel.writeString(time);
        parcel.writeString(repeat);
        parcel.writeString(status);
    }
    public static final Creator<Item> CREATOR = new Creator<Item>() {
        @Override
        public Item createFromParcel(Parcel in) {
            return new Item(in);
        }

        @Override
        public Item[] newArray(int size) {
            return new Item[size];
        }
    };
}

File: app/src/main/java/com/OverSadBoy/samurairise/model/database/Item.kt
package com.OverSadBoy.samurairise.model.database
import android.os.Parcel
import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity
class Item : Parcelable {

    @PrimaryKey
    var id: Int = 0
    var time: String? = null
    var repeat: String? = null
    var status: String? = null

    constructor(id: Int, time: String, repeat: String, status: String) {
        this.id = id
        this.time = time
        this.repeat = repeat
        this.status = status
    }

    private constructor(parcel: Parcel) {
        id = parcel.readInt()
        time = parcel.readString()
        repeat = parcel.readString()
        status = parcel.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeInt(id)
        parcel.writeString(time)
        parcel.writeString(repeat)
        parcel.writeString(status)
    }

    companion object CREATOR : Parcelable.Creator<Item> {
        override fun createFromParcel(parcel: Parcel): Item {
            return Item(parcel)
        }

        override fun newArray(size: Int): Array<Item?> {
            return arrayOfNulls(size)
        }
    }
}
